### ISSUE TITLE
Payload is missing <dynamicTemplateData> value when testing <send> function

### ISSUE THREAD

**Issue by eventgo-admin** [sendgrid/sendgrid-nodejs#1214]
### Issue Summary
When I use the following mock structure with `nock` (**A**) and I call it in the following way (**B**), it works, but here is the problem. The post method is missing the `dynamicTemplateData` value (used in **B**). Am I missing additional API call or I do something wrong?

The actual code works and the template is sent with the correct value for the `link`.

### Code Snippet

**A**

```
nock(sendGridHost)
        .defaultReplyHeaders({
          "access-control-allow-origin": "*",
          "access-control-allow-headers": "User-Agent, Authorization",
        })
        .options("/v3/mail/send")
        .reply(204)
        .post(
          "/v3/mail/send",
          {
            from: { email: emailFrom },
            personalizations: [
              { to: [{ email: receiverEmail }] },
            ],
            template_id: sendGridTemplateId,
          },
          {
            reqheaders: {
              authorization: `Bearer ${sendGridApiKey}`,
            },
          }
        )
        .reply(202);
```

**B**

``` 
const result = await sendGridEmail.send({
    from: emailFrom ,
    to: receiverEmail ,
    templateId: emailTemplateId,
    dynamicTemplateData: {
      link: myLink,
    },
  });
```

### Technical details:
* sendgrid-nodejs version: 7.3.0
* node version: 12.16.1

**Comment by eventgo-admin** [comment]
If I change any part of the nock expectation,  it complains that it was not called. So i am guessing the setup is correct.

**Comment by childish-sambino** [comment]
Odd. Have you tried printing/logging the request body to ensure it includes the link: https://github.com/sendgrid/sendgrid-nodejs/blob/main/TROUBLESHOOTING.md#viewing-the-request-body

**Comment by eventgo-admin** [comment]
Hey, sorry for the late reply

this is what I get when I add the code from the link

```
{
    "from":{"email":"test@test.test"},
    "personalizations":[
        {
        "to":[{"email":"testreceived@test.test"}],
        "dynamic_template_data": {"link":"test_link_value"}
        }
    ],
    "template_id":"test_template_id"
}
```

**Comment by thinkingserious** [comment]
Hello @eventgo-admin,

In example A, it does not look like you have defined the `dynamic_template_data` field, which should be in the `personalizations` object.

With best regards,

Elmer

**Comment by eventgo-admin** [comment]
@thinkingserious 
If I define the field in `nock` as follows
```
personalizations: [
    { 
        to: [{ email: receiverEmail }],
        dynamic_template_data: { link: test_link_value }
    },
],
```

It fails because `nock` cannot find mock of the request. 
``` Error: Nock: No match for request```
If I remove the `dynamic_template_data` nock works

**Comment by eventgo-admin** [comment]
OK I found the issue.
For everyone who has the same problem as me, here is the explanation:

**TL;DR**
in tests - use tempalte id that starts with ```d-``` otherwise it does not set ```dynamic_template_data``` key to the request

**Long answer**

When new ```Mail``` class is created, there is this call in the constructor ```fromData(data)``` which breaks the data passed into the required fields. In that function you have a call for ```this.setTemplateId(templateId);``` which (and here is the important part) sets the ```templateId```. 

And here is the important part, also it sets ```isDynamic=true``` ONLY IF  the ```templateId``` that you passed starts with ```d-```.

So after that in the form data you have 
```
    if (this.isDynamic) {
      this.setDynamicTemplateData(dynamicTemplateData);
    } else {
      this.setSubstitutions(substitutions);
      this.setSubstitutionWrappers(substitutionWrappers);
    }
```
So it will populate the ```dynamic_template_data``` only if the ```isDynamic``` is ```true``` and this will happen only if ```templateId``` you passed starts with the ```d-```
